modules/core/04-channel/keeper/upgrade.go

```diff
@@ -668,23 +667,48 @@ func (k Keeper) ChanUpgradeTimeout(
 		)
 	}
 
+	// proof height must be from a height after timeout has elapsed.
+	if !upgrade.Timeout.Height.IsZero() && proofHeight.LT(upgrade.Timeout.Height) {
+		return errorsmod.Wrap(types.ErrInvalidUpgradeTimeout, "timeout height is not valid")
+	}
+
 	// proof must be from a height after timeout has elapsed. Either timeoutHeight or timeoutTimestamp must be defined.
 	// if timeoutHeight is defined and proof is from before timeout height, abort transaction
 	proofTimestamp, err := k.connectionKeeper.GetTimestampAtHeight(ctx, connection, proofHeight)
 	if err != nil {
 		return err
 	}
 
-	timeout := upgrade.Timeout
-	proofHeightIsInvalid := timeout.Height.IsZero() || proofHeight.LT(timeout.Height)
-	proofTimestampIsInvalid := timeout.Timestamp == 0 || proofTimestamp < timeout.Timestamp
-	if proofHeightIsInvalid && proofTimestampIsInvalid {
-		return errorsmod.Wrap(types.ErrInvalidUpgradeTimeout, "timeout has not yet passed on counterparty chain")
+	// if timeout timestamp is defined then the consensus time
+	// from proof height must be greater than timeout timestamp.
+	if upgrade.Timeout.Timestamp != 0 && proofTimestamp < upgrade.Timeout.Timestamp {
+		return errorsmod.Wrap(types.ErrInvalidUpgradeTimeout, "timeout timestamp is not valid")
 	}
 
-	// counterparty channel must be proved to still be in OPEN state or INITUPGRADE state (crossing hellos)
-	if !collections.Contains(counterpartyChannel.State, []types.State{types.OPEN, types.INITUPGRADE}) {
-		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected one of [%s, %s], got %s", types.OPEN, types.INITUPGRADE, counterpartyChannel.State)
+	// counterparty channel must be proved to still be in OPEN state or FLUSHING state.
+	if !collections.Contains(counterpartyChannel.State, []types.State{types.OPEN, types.STATE_FLUSHING}) {
```

Comment:
`ValidateBasic` is currently only allowing `OPEN` for counterparty channel state

Reply:
I guess if it's in ValidateBasic it's not actually needed here, I think it's no harm leaving this in to show intent though.